### PR TITLE
Fix sonar warnings

### PR DESCRIPTION
**Pull request by dundich**
Signed-off-by: krivchenko_kv <kvkrivchenko@activebc.ru>

### CODE REVIEW

**Comment by zarusz** [comment]
@dundich  looks good. Can you please squash all your commits into one?
Then I can accept the PR.
Thanks for your contribution!

**Comment by dundich** [comment]
fixed: all commits in one... 
but he didn't pass the intergation test, I don't know :(

**Comment by zarusz** [comment]
@dundich I believe the branch you want to merge from your fork does not have the recent changes needed to bring Kafka integration tests back on track.

From this view, your branch is 3 commits behind: https://github.com/dundich/SlimMessageBus/compare/sonar-fix...zarusz%3ASlimMessageBus%3Amaster

Please pull the latest changes from this (original) repo into your fork.
Then rebase your branch aginst the latest master.

**Comment by dundich** [comment]
Thanks for the tips. Done!

File: src/SlimMessageBus.Host/Consumer/MessageScopeWrapper.cs
Comment (dundich) [inline src/SlimMessageBus.Host/Consumer/MessageScopeWrapper.cs:1]: Removed unnecessary dependencies from the bloated constructor with logging

File: src/SlimMessageBus.Host.Configuration/Builders/MessageBusBuilder.cs [hunk 1]
```diff
@@ -334,7 +334,7 @@ public MessageBusBuilder AddChildBus(string busName, Action<MessageBusBuilder> b
             child.MergeFrom(Settings);
         }
 
-        builderAction?.Invoke(child);
+        builderAction.Invoke(child);
```
Comment (zarusz) [inline src/SlimMessageBus.Host.Configuration/Builders/MessageBusBuilder.cs:337]: Remove invoke

**Comment by dundich** [comment]
## Unused local variables should be removed ???

https://github.com/zarusz/SlimMessageBus/blob/bd33379966f710a13b5f4fc15ac084b78ab3eb08/src/SlimMessageBus.Host/DependencyResolver/ServiceCollectionExtensions.cs#L76            

 ```
    var messageBusSettings = svp.GetRequiredService<MessageBusSettings>();
```

ps. I didn't remove it because I'm afraid of the side effect.

**Comment by dundich** [comment]
## Constructors should only call non-overridable methods

https://github.com/zarusz/SlimMessageBus/blob/bd33379966f710a13b5f4fc15ac084b78ab3eb08/src/SlimMessageBus.Host/Hybrid/HybridMessageBus.cs#L40C11-L40C12

https://rules.sonarsource.com/csharp/RSPEC-1699/

```
    protected virtual MessageBusBase BuildBus(MessageBusBuilder builder)
```

ps. Maybe - sealed class?

**Comment by dundich** [comment]
That's it. I'm done. :)

**Comment by zarusz** [comment]
> ## Unused local variables should be removed ???
> https://github.com/zarusz/SlimMessageBus/blob/bd33379966f710a13b5f4fc15ac084b78ab3eb08/src/SlimMessageBus.Host/DependencyResolver/ServiceCollectionExtensions.cs#L76
> 
> ps. I didn't remove it because I'm afraid of the side effect.

It should be safe to remove - I will do it. 


Thanks for contribution @dundich !

**Pull request merged**